### ISSUE TITLE
dump_flags() and geomeTRIC optimizer

### ISSUE THREAD

**Issue by mkrompiec** [ajz34/vdw#1]
Thanks for making this code available. I noticed a weird issue when using your wrapper together with geomeTRIC optimiser: the code crashes due to parse_config not seeing the parameters. The workaround is to disable dump_flags(), i.e. to put a return statement in the 1st line of dump_flags.

**Comment by ajz34** [comment]
Hi :smile: 

Actually I'm not sure how to reproduce this problem. Could you share some more information on this problem?
I made some attempt to reproduce this issue; but the following code (in the following collapsed section) seems work on my computer. (I'm not sure whether the following result is physically meaningful, just know that the code runs)
<details>

<summary>code to optimize with self-defined parameter of DFT-D4 by geomeTRIC</summary>

```python
from pyscf import gto, dft
from vdw import to_dftd4
from pyscf.geomopt.geometric_solver import optimize


def job_of_dftd4():
    mol = gto.Mole(atom="""
    O  0.  0.     0.
    H  0. -0.137  0.981
    H  0.  0.981 -0.137
    O  0.  0.     2.
    H  0. -0.137  2.981
    H  0.  0.981  1.863
    """, basis="6-31G").build()
    # DFTD4 default bj-eeq-atm version
    mf = to_dftd4(dft.RKS(mol, xc="PBE"), param={"s8": 4.0, "a1": 0.1, "a2": 3.2}).run()
    print(mf.with_vdw._param_output)
    print(mf.e_vdw)

    def callback(dct):
        print(dct["self"].scanner.base.with_vdw._param_output)
        return
    mol_opt = optimize(mf, callback=callback)
    print("== Optimized Molecule ==")
    print(mol_opt.atom_coords(unit="Angstrom"))
    # final energy for {"s8": 4.0, "a1": 0.1, "a2": 3.2}
    #     -152.6893725099
    # final energy for default parameter of PBE, DFT-D4
    #     -152.6156118516


if __name__ == '__main__':
    print()
    print("=== obtain_ts_vdw_from_mf ===")
    job_of_dftd4()
```
</details>

**Comment by mkrompiec** [comment]
Thanks for looking into this. I have just tried to reproduce it, on another VM, using pyscf 2.1.1 and 2.2.0, as well as python 3.10 and 3.11. To my great surprise, I can't reproduce the error (tried several test cases). Must have been something specific to my local pyscf build and/or python environment on the other machine.
The only issue I had was that `tomli` (or some other TOML library) should be specified as a dependency.

**Comment by ajz34** [comment]
@mkrompiec Sorry, I just found this problem when working on other codes hours ago >.< You were right and thanks of pointing this out.
When declaring verbose>=4 in molecule definition `gto.Mole(..., verbose=4)`, then exception raised from `dump_flags`.
This is probably a problem of poor inheritance design. I made a bug-fix in [437e7c8](https://github.com/ajz34/vdw/commit/437e7c8ae6a1caaddd4b876b26830866287595ed) and you may try it out.

**Comment by mkrompiec** [comment]
Thank you! This fixes the problem.
